### PR TITLE
Allow customization of spaceline additional segments.

### PR DESCRIPTION
**Pull request by riccardomurri**
Introduce a new customization variable
`spacemacs-spaceline-additional-segments', which is a list of the
additional segments that should be inserted in the modeline when it is
initialized.

(This is basically the same patch as PR #6846 but applied to the "develop"
branch instead of "master".)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,6 +18,7 @@ my-org/
 org-files/
 semanticdb/
 edts/
+/recentf
```
Comment (d12frosted) [inline .gitignore]: This change should not be part of this PR 😸

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop (at least, the relevant commit). You can safely delete your branch.
